lib/matplotlib/dviread.py

```diff
@@ -854,47 +881,68 @@ def _register(self, words):
         There is some difference between <foo.pfb and <<bar.pfb in
         subsetting, but I have no example of << in my TeX installation.
         """
-
         # If the map file specifies multiple encodings for a font, we
         # follow pdfTeX in choosing the last one specified. Such
         # entries are probably mistakes but they have occurred.
         # http://tex.stackexchange.com/questions/10826/
         # http://article.gmane.org/gmane.comp.tex.pdftex/4914
 
-        texname, psname = words[:2]
-        effects, encoding, filename = '', None, None
-        for word in words[2:]:
-            if not word.startswith('<'):
-                effects = word
-            else:
-                word = word.lstrip('<')
-                if word.startswith('[') or word.endswith('.enc'):
+        empty_re = re.compile(br'%|\s*$')
+        word_re = re.compile(
```

Comment:
I learned quite a bit about regular expressions understanding this pattern.

For example, you can make them readable and name groups!